libs/vkd3d/resource.c

```diff
@@ -711,53 +707,10 @@ static HRESULT vkd3d_get_image_create_info(struct d3d12_device *device,
         image_info->pQueueFamilyIndices = NULL;
     }
 
-    if (heap_properties && is_cpu_accessible_heap(heap_properties))
-    {
-        image_info->initialLayout = VK_IMAGE_LAYOUT_PREINITIALIZED;
-        /* Required for ReadFromSubresource(). */
-        image_info->tiling = VK_IMAGE_TILING_LINEAR;
-
-        if ((vkd3d_config_flags & VKD3D_CONFIG_FLAG_IGNORE_RTV_HOST_VISIBLE) &&
-                (image_info->usage & VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT))
-        {
-            WARN("Workaround applied. Ignoring RTV on linear resources.\n");
-            image_info->usage &= ~VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT;
-            if (resource)
-                resource->desc.Flags &= ~D3D12_RESOURCE_FLAG_ALLOW_RENDER_TARGET;
-        }
-    }
-    else
-    {
-        image_info->initialLayout = VK_IMAGE_LAYOUT_UNDEFINED;
-    }
-
     if ((image_info->flags & VK_IMAGE_CREATE_MUTABLE_FORMAT_BIT) &&
             !vkd3d_format_check_usage_support(device, format->vk_format, image_info->usage, image_info->tiling))
         image_info->flags |= VK_IMAGE_CREATE_EXTENDED_USAGE_BIT;
 
-    if (image_info->tiling == VK_IMAGE_TILING_LINEAR)
-    {
-        bool supported = vkd3d_is_linear_tiling_supported(device, image_info);
```

Comment:
Do we remember to remove this function? It should be dead code at this point.

Reply:
In a similar vein, do we fix up memory_info for host visible to only consider that we need to place VkBuffers?